animations/test/misc/hero_animation_test.dart

```diff
@@ -0,0 +1,109 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:animations/src/misc/hero_animation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+Widget createHeroAnimationDemoScreen() => MaterialApp(
+      home: HeroAnimationDemo(),
+    );
+
+void main() {
+  group('Hero Animation Tests', () {
+    testWidgets('Size of Container changes on Tap', (tester) async {
+      await tester.pumpWidget(createHeroAnimationDemoScreen());
+
+      // Get the initial Container.
+      final initialContainer =
+          tester.firstWidget(find.byType(Container)) as Container;
+
+      // Get initial height of Container.
+      // In this case minHeight is equal to maxHeight.
+      var initialHeight =
+          initialContainer.constraints.heightConstraints().maxHeight;
+
+      // Get initial width of Container.
+      // In this case minWidth is equal to maxWidth.
+      var initialWidth =
+          initialContainer.constraints.widthConstraints().maxWidth;
+
+      // Calculate initial area of Container
+      var initialArea = initialHeight * initialWidth;
+
+      // Tap on the GestureDetector.
+      await tester.tap(find.byType(GestureDetector));
+      await tester.pumpAndSettle();
+
+      // Get the final Container.
+      final finalContainer =
+          tester.firstWidget(find.byType(Container)) as Container;
+
+      // Get height of Container.
+      // In this case minHeight is equal to maxHeight.
+      var finalHeight =
+          finalContainer.constraints.heightConstraints().maxHeight;
+
+      // Get width of Container.
+      // In this case minWidth is equal to maxWidth.
+      var finalWidth = finalContainer.constraints.widthConstraints().maxWidth;
+
+      // Calculate final area of Container
+      var finalArea = finalHeight * finalWidth;
+
+      expect(
+        initialArea,
+        lessThan(finalArea),
+      );
+    });
+
+    testWidgets('Color of Container changes on Tap', (tester) async {
```

Comment:
This test isn't just checking that the color changes. It's checking that it starts as a specific color and ends up being another specific color. Why not try simply comparing the before and after colors to see if they match?

Reply:
Yes sure, fixing this in the next commit.